programs/ssl/ssl_server2.c

```diff
@@ -4287,8 +4288,8 @@ int main( int argc, char *argv[] )
     sni_free( sni_info );
 #endif
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
-    if( ( ret = psk_free( psk_info ) ) != 0 )
-        mbedtls_printf( "Failed to list of opaque PSKs - error was %d\n", ret );
+    if( ( psk_free_ret = psk_free( psk_info ) ) != 0 )
+        mbedtls_printf( "Failed to list of opaque PSKs - error was %d\n", psk_free_ret );
```

Comment:
I'd say that it is rather that we want to avoid the printf in case of failure rather than preserving `ret`, thus rather:
```
ret = psk_free( psk_info );
if( ( ret != 0 ) && ( opt.query_config_mode == DFL_QUERY_CONFIG_MODE ) )
    mbedtls_printf ...
```

Reply:
Do you mean that all printf after the `exit` label should be changed to be dependent from `query_config_mode`?

Reply:
That is my understanding of the solution that we end up with. Before this PR there was just a return thus a return from the function without any output. Thus to keep the same visible behavior there shouldn't be any output when we go through the code after the exit label. Does it make sense?

Reply:
Absolutely.